### DISCUSSION TITLE
Connect Controller With LAN/ETHERNET cable

### DISCUSSION THREAD

**Discussion by hanifhan** [grblHAL/core#275] [Q&A] [answered]
hi , i want to connect my controller with ETHERNET , is there any configuration that i should set for it ? and what is the IP address of the ETHERNET ?

**Comment by terjeio** [comment]
Just enable ethernet and the required protocols (telnet and/or websocket for streaming gcode) in _my_machine.h_ and reflash. Default is to use DHCP for assigning an IP address, `$I` will output it alongside other info. If you want to use a static IP then you have to [configure that](https://github.com/grblHAL/core/wiki/Additional-or-extended-settings#network-interface-settings) to match your network.

A reset of plugin settings (`$RST=&`) might be required to get the default settings right.

**Reply by hanifhan** [reply]
im sorry terje , im not quite understand , but Im using STM32F446xx driver , and i use W5500 module and  i dont seem to find enet.h and etc , where can i find it ? or is there anyway to activate it with stm32f446 ?

**Reply by terjeio** [reply]
The only way to get ethernet for STM32F4xx is to write the neccesary code for it. A lwIP driver for W5500 and _enet.c_ is required.
For now your only options are the [drivers that has networking support](https://github.com/grblHAL/Plugin_networking#driver-support).

**Reply by terjeio** [reply, marked as answer]
The STM32F4xx driver has gotten support for W5100S (untested) and W5500 breakouts. SPI pins has to be defined in the map file for it to work - like in [this map](https://github.com/grblHAL/STM32F4xx/blob/c06947d825e3cbd96a4c0b4f204ad30a7e01a2a6/Inc/st_morpho_map.h#L225-L234).

**Comment by andrewmarles** [comment]
Already have this working on FlexiHAL with just re-mapping the SPI pins.  So awesome, the new G65 macro support is amazing.
I was able to test streaming gcode from SD while connected over websockets and it seemed to work fine, still need to get it on a machine.

![image](https://github.com/grblHAL/core/assets/6061539/dbb73d5d-a7f5-4dcd-b8ba-3ca7803b6a5e)
